runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/fn/control/BeamFnMapTaskExecutor.java

```diff
@@ -423,32 +420,28 @@ private CounterUpdate monitoringInfoToCounterUpdate(MonitoringInfo monitoringInf
 
         String type = monitoringInfo.getType();
 
-        // todomigryz: run MonitoringInfo through validation process.
-        // refer to https://github.com/apache/beam/pull/6799
-
+        // todo(migryz): run MonitoringInfo through Proto validation process.
+        // Requires https://github.com/apache/beam/pull/6799 to be merged.
         if (urn.startsWith(BEAM_METRICS_USER_PREFIX)) {
           if (!type.equals("beam:metrics:sum_int_64")) {
-            LOG.warn(
+            throw new RuntimeException(
                 "Ignoring user-counter MonitoringInfo with unexpected type."
```

Comment:
These error messages don't quite make sense for throwing exceptions. We're no longer ignoring them, we're failing execution. Suggestion:

```java
throw new RuntimeException("Encountered MonitoringInfo with unexpected type. Expected [..]")
```

Reply:
I pushed a commit with this fixup.